distribution/tools/java-version-checker/build.gradle

```diff
@@ -1,12 +1,27 @@
-apply plugin: 'elasticsearch.build'
+apply plugin: 'elasticsearch.java'
 
-targetCompatibility = JavaVersion.VERSION_1_7
+sourceSets {
+  unsupportedJdkVersionEntrypoint
+}
+
+tasks.named('compileUnsupportedJdkVersionEntrypointJava').configure {
```

Comment:
FYI, we can use `sourceSets.unsupportedJdkVersionEntrypoint.compileJavaTaskName` here.